lib/simplecov/result_merger.rb

```diff
@@ -87,6 +94,25 @@ def store_result(result)
         end
         true
       end
+
+    private
+
+      # Ensure only one process is reading or writing the resultset at any
+      # given time
+      def synchronize_resultset
+        # make it reentrant
+        return yield if defined?(@resultset_locked) && @resultset_locked
```

Comment:
Which works as we are just one process, got it, thanks for taking the time to explain! :)